### PR TITLE
#5 refactor

### PR DESCRIPTION
**Pull request by UdjinSkobelev**

### CODE REVIEW

File: core/src/main/java/org/webtree/auth/repository/AuthRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Override
+    public Optional<User> findByUsername(String name) {
+      return   Optional.ofNullable(operations.selectOne(Query.query(where("username").is(name)), User.class));
+    }
+
+    @Override
+    public User saveIfNotExists(User user) {
+//        operations.insert()
```
Comment (Max-Levitskiy) [inline core/src/main/java/org/webtree/auth/repository/AuthRepositoryImpl.java]: Remove comment

File: core/src/main/resources/messages.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
+login.badCredentials=Incorrect username or password.
+social.linked.current=You already have connect this account.
+social.linked.another=This account was connected by someone else.
+social.provider.notSupported=Current provider not supported.
+social.invalidToken=Invalid OAuth access token
+incorrect.data=The data you have passed is incorrect.
+social.data.save.exception=Cant save data.
```
Comment (Max-Levitskiy) [inline core/src/main/resources/messages.properties]: Can't :)

File: core/src/test/java/org/webtree/auth/controller/AuthControllerTest.java [hunk 3]
```diff
@@ -45,12 +47,25 @@ public void setUp() {
 
     @Test
     void shouldReturn2XxOkIfUserDoesNotExist() throws Exception {
+        when(service.registerIfNotExists(any(AuthDetails.class))).thenReturn(true);
         mockMvc
-                .perform(post("/rest/user/register")
+                .perform(post("/rest/user/registerIfNotExists")
```
Comment (Max-Levitskiy) [inline core/src/test/java/org/webtree/auth/controller/AuthControllerTest.java]: Do we have another option? :)

File: core/src/test/java/org/webtree/auth/controller/AuthControllerTest.java [hunk 4]
```diff
@@ -45,12 +47,25 @@ public void setUp() {
...
                         .contentType(MediaType.APPLICATION_JSON)
                         .content(objectMapper.writeValueAsString(authDetails)))
                 .andExpect(status().isCreated());
-        verify(service).register(authDetailsWithEncodedPassword);
+        verify(service).registerIfNotExists(authDetailsWithEncodedPassword);
+    }
+
+    @Test
+    void shouldReturn4XxBadErquestIfUserExist() throws Exception {
```
Comment (Max-Levitskiy) [inline core/src/test/java/org/webtree/auth/controller/AuthControllerTest.java]: ...BadRequest...

File: core/pom.xml [hunk 5]
```diff
@@ -2,92 +2,176 @@
...
         <dependency>
             <groupId>org.assertj</groupId>
             <artifactId>assertj-core</artifactId>
-            <scope>test</scope>
+            <version>${assert-j-core.version}</version>
         </dependency>
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-junit-jupiter</artifactId>
-            <scope>test</scope>
```
[the remark this answers is not in the document]
Comment (UdjinSkobelev) [reply core/pom.xml]: mistake

File: core/src/main/java/org/webtree/auth/config/EmbeddedCassandraConfig.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import java.io.IOException;
+
+/**
+ * Created by Udjin Skobelev on 03.12.2018.
+ */
+
+@Profile("dev")
+@Configuration
+public class EmbeddedCassandraConfig extends CassandraAuthConfiguration {
+    private static final String MESSAGE = "Cant start embedded cassandra server";
```
[the remark this answers is not in the document]
Comment (UdjinSkobelev) [reply core/src/main/java/org/webtree/auth/config/EmbeddedCassandraConfig.java]: +

File: core/src/main/java/org/webtree/auth/repository/AuthRepositoryImpl.java (hunk 1 above)
Comment (UdjinSkobelev) [reply core/src/main/java/org/webtree/auth/repository/AuthRepositoryImpl.java]: +

File: core/src/main/resources/messages.properties (hunk 2 above)
Comment (UdjinSkobelev) [reply core/src/main/resources/messages.properties]: +

File: core/src/test/java/org/webtree/auth/controller/AuthControllerTest.java (hunk 3 above)
Comment (UdjinSkobelev) [reply core/src/test/java/org/webtree/auth/controller/AuthControllerTest.java]: idea's refactoring)

File: core/src/test/java/org/webtree/auth/controller/AuthControllerTest.java (hunk 4 above)
Comment (UdjinSkobelev) [reply core/src/test/java/org/webtree/auth/controller/AuthControllerTest.java]: +

File: auth-core/pom.xml [hunk 7]
```diff
@@ -2,92 +2,176 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-test</artifactId>
+            <version>${spring-core.version}</version>
+        </dependency>
         <dependency>
             <groupId>org.assertj</groupId>
             <artifactId>assertj-core</artifactId>
-            <scope>test</scope>
```
[the remark this answers is not in the document]
Comment (UdjinSkobelev) [reply auth-core/pom.xml]: +++

File: auth-core/src/main/java/org/webtree/auth/config/AuthConfigurationProperties.java [hunk 8]
```diff
@@ -71,9 +58,9 @@ public PasswordEncoder getEncoder() {
     public static class Route {
 
         /**
-         * Endpoint for register user.
+         * Endpoint for registerIfNotExists user.
          */
-        private String register = "/rest/user/register";
+        private String register = "/rest/user/registerIfNotExists";
```
[the remark this answers is not in the document]
Comment (UdjinSkobelev) [reply auth-core/src/main/java/org/webtree/auth/config/AuthConfigurationProperties.java]: reverted)

File: auth-core/src/test/java/org/webtree/auth/extension/EmbeddedCassandraExtension.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+import org.cassandraunit.utils.EmbeddedCassandraServerHelper;
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class EmbeddedCassandraExtension implements BeforeAllCallback, AfterTestExecutionCallback {
+
+    @Override
+    public void beforeAll(ExtensionContext extensionContext) throws Exception {
+        EmbeddedCassandraServerHelper.startEmbeddedCassandra(60000L);
```
[the remark this answers is not in the document]
Comment (UdjinSkobelev) [reply auth-core/src/test/java/org/webtree/auth/extension/EmbeddedCassandraExtension.java]: We can use Embedded Cassandra config for it.

File: auth-core/pom.xml [hunk 10]
```diff
@@ -2,92 +2,176 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-    <parent>
```
[the remark this answers is not in the document]
Comment (UdjinSkobelev) [reply auth-core/pom.xml]: reverted

File: auth-core/src/main/resources/messages.properties [hunk 11]
```diff
@@ -0,0 +1,8 @@
+login.badCredentials=Incorrect username or password.
+social.linked.current=You already have connect this account.
+social.linked.another=This account was connected by someone else.
+social.provider.notSupported=Current provider not supported.
+social.invalidToken=Invalid OAuth access token
+incorrect.data=The data you have passed is incorrect.
+social.data.save.exception=Cant save data.
```
Comment (Max-Levitskiy) [inline auth-core/src/main/resources/messages.properties]: cant... :)

File: auth-core/src/main/java/org/webtree/auth/config/WebConfig.java [hunk 12]
```diff
@@ -19,4 +19,6 @@
     public void addArgumentResolvers(List<HandlerMethodArgumentResolver> resolvers) {
         resolvers.add(passwordProcessor);
     }
+
+    //todo запускать спринг бут
```
Comment (Max-Levitskiy) [inline auth-core/src/main/java/org/webtree/auth/config/WebConfig.java]: Remove it :)

File: auth-core/src/main/java/org/webtree/auth/domain/User.java [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+    @Override
+    public boolean isEnabled() {
+        return true;
+    }
+
+    public String getId() {
+        return this.id;
+    }
+
+    public @Indexed String getUsername() {
```
[the remark this answers is not in the document]
Comment (UdjinSkobelev) [reply auth-core/src/main/java/org/webtree/auth/domain/User.java]: ok)

File: auth-core/src/main/java/org/webtree/auth/config/WebConfig.java (hunk 12 above)
Comment (UdjinSkobelev) [reply auth-core/src/main/java/org/webtree/auth/config/WebConfig.java]: ok

**Pull request merged**